gammapy/utils/nddata.py

```diff
@@ -0,0 +1,441 @@
+"""Scratch for NDDataArray class"""
+
+import itertools
+import numpy as np
+from astropy.units import Quantity, Unit
+from astropy.table import Table, Column
+
+
+class NDDataArray(object):
+    """ND Data Array
+
+    This class represents an ND Data Array. The data stored as numpy array
+    attribute. The data axis are separate classes and this class has them as
+    members. The axis order follows numpy convention for arrays, i.e. the axis
+    added last is at index 0. For an example see NDData_demo.ipynb.
```

Comment:
How do you want to handle docs for this?
Maybe for now copy the `NDData_demo.ipynb` to gammapy-extra and maintain it there while this class develops (as the option requiring least effort for now)?